### ISSUE TITLE
[Feature Request] Implement TQC for the example algorithms

### ISSUE THREAD

**Issue by maxweissenbacher** [pytorch/rl#1623]
## Motivation

I suggest adding an implementation of [TQC](https://github.com/SamsungLabs/tqc_pytorch) to the examples. I suggest adding this as a request in the 'call for distributions' stack. I would be happy to take on the implementation.

## Solution

Add a performant, clear and minimal implementation of TQC to the examples. I would base the implementation on the already existing implementation of SAC in the examples. This seems reasonable due to the overall similar structure of the algorithms. In the [Stable-Baselines3 library implementation of TQC](https://sb3-contrib.readthedocs.io/en/master/modules/tqc.html#comments), the implementation is likewise based on their corresponding implementation of SAC.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] Add a request to the [call for contributions](https://github.com/pytorch/rl/issues/509).
- [x] Adapt the [implementation of SAC from the examples](https://github.com/pytorch/rl/blob/main/examples/sac/sac.py) to TQC
- [ ] Benchmark?

```[tasklist]
### Tasks
```

**Comment by vmoens** [comment]
Hi @maxweissenbacher 
Adding TQC would be fantastic and greatly appreciated!
Any help we can provide, just let us know. I'm adding this to the call for contributions straight away!

**Comment by maxweissenbacher** [comment]
Hi @vmoens, that's great, thank you! I'll get started right away and come back here in case I need assistance (more likely than not).

**Comment by maxweissenbacher** [comment]
I cannot seem to get the logger to work for me. By running the (unchanged) training script for the SAC example, trying to execute the`get_logger` function gives me the following error:

`wandb: WARNING Path sac_logging/wandb/ wasn't writable, using system temp directory.`

Any suggestions?

**Comment by vmoens** [comment]
For the logger it seems you lack write access in the folder where you're executing the code.

For the OSS contribution:
- make a fork of torchrl on your username
- write code in your separate branch locally on your machine 
- when you're at a stage where you feel the code is mature and/or you want to show it to us to get feeeback, commit the changes on that branch and push it to your fork. Then create a PR from that branch towards main on the PyTorch org.

You will need to format the code etc for which I invite you to read the CONTRIBUTE.md file in torchrl.

Thanks again for your help!

**Comment by maxweissenbacher** [comment]
Thank you for all this info. I have submitted a pull request now, hopefully I did it right!

I couldn't get the logging to work, so I implemented a custom logger to use on my local machine instead - however the logging is just as in the other example files in the pull request, so it should hopefully work as expected for you.

**Comment by chinu0609** [comment]
hi @vmoens I am new here I want to contribute could you point me to any beginner friendly issue, or something in your mind thank you.

**Comment by gtnv** [comment]
I think this needs to be done and put to rest. I will make the PR today. Claiming ownership.

**Comment by vmoens** [comment]
You can use [forward slash]assign

**Comment by gtnv** [comment]
nice. didnt know that was available here.

**Comment by gtnv** [comment]
cc @theap06 @vmoens Im done implementing TQC and have some updates to share.

The PRs are in order of merge-priority.
[[BugFix] [TQC] [1/3] Support list-valued critics in SoftUpdate
](https://github.com/pytorch/rl/pull/4077)

[[Feature] [TQC] [2/3] Add truncated quantile critics
](https://github.com/pytorch/rl/pull/4078)

[[Test] [TQC] [3/3] Add TQC numerical contracts test
](https://github.com/pytorch/rl/pull/4079)

While testing my work (across all 3) on h100s, I found 2 old bugs resurfacing as blockers for a functional implementation. The first was this one: [[BUG] SoftUpdate fails for list-valued SAC critics](https://github.com/pytorch/rl/issues/4076) . Its a leftover bug exposed by an old bugfix. The second is [[BUG] Numerical Instability issues with torchrl.modules.TanhNormal  ](https://github.com/pytorch/rl/issues/2199) This second bug is linked to loss behavior. evidence in #4078 . (I didnt observe the exact same bug, but codex told me this is the issue for it. codex is wrong so I will probably make a new issue and pr to fix). 

I have a fix up for the first problem. will draft second fix. Thanks as always guys.

**Comment by theap06** [comment]
@gtnv thanks! will review asap

**Comment by gtnv** [comment]
@theap06 epic. btw the solution to the tanhnormal bug is here: https://github.com/pytorch/rl/pull/4080

**Comment by gtnv** [comment]
btw @vmoens this can be closed
